datamodel.itop-standard-email-synchro.xml

```diff
@@ -404,7 +404,33 @@ EOF
 	{
 		// First check if there is any iTop object mentioned in the headers of the eMail
 		$oTicket = parent::GetRelatedTicket($oEmail);
-		
+
+		if ($oTicket === null && MetaModel::GetModuleSetting('itop-standard-email-synchro', 'aggregate_answers')!==false) {
+			if (array_key_exists('in-reply-to', $oEmail->aHeaders)) {
+				$sInReplyToMsgId = $oEmail->aHeaders['in-reply-to'];
+				$sOQL = "SELECT EmailReplica WHERE message_id = :in_reply_to";
+				$oSet = new DBObjectSet(DBObjectSearch::FromOQL($sOQL), array(), array('in_reply_to' => $sInReplyToMsgId));
+				$oSet->OptimizeColumnLoad(['EmailReplica'=>['ticket_id']]);
+				$iEmailReplicaCount = $oSet->Count();
+				if ($iEmailReplicaCount > 0){
+					/** @var EmailReplica $oEmailReplica */
+					$oEmailReplica = $oSet->Fetch();
+					if ($iEmailReplicaCount > 1) {
+						$this->Trace("iTop Simple Email Synchro: WARNING the message ({$oEmail->sUIDL}) replies to the message_id=".$sInReplyToMsgId."
+						This message_id corresponds to $iEmailReplicaCount email replica in iTop.");
+					}
+					$oTicket = MetaModel::GetObject('Ticket', $oEmailReplica->Get('ticket_id'), false);
+					if ($oTicket === null) {
+						$this->Trace("iTop Simple Email Synchro: WARNING the message ({$oEmail->sUIDL}) got a EmailReplica#{$oEmailReplica->GetKey()}
+						with the ticket_id={$oEmailReplica->Get('ticket_id')} but the ticket does not exist.");
+					} elseif ($oTicket->Get("operational_status") === "closed") {
+						$this->Trace("iTop Simple Email Synchro: WARNING the message ({$oEmail->sUIDL}) is related to the CLOSED ticket {$oTicket->Get('id')}");
+						$oTicket = null;
+					}
```

Comment:
ajoute:
else {
 return $oTicket;
}

ça rend le code plus facile à lire. et tu peux virer tous les "$oTicket = null;"

Reply:
Why should we remove this "$oTicket = null;" ? It's necessary to indicate that we don't want to use the current (closed) ticket.

Reply:
the sooner you return found ticket the simpler code will be afterwhile. you dont have to check whether oTicket is null or not. either object has been returned or not.